packages/@aws-cdk/region-settings/lib/region.ts

```diff
@@ -0,0 +1,186 @@
+import fs = require('fs');
+import path = require('path');
+import { DefaultServicePrincipalMaker, IServicePrincipalMaker } from './service-principal';
+
+const REGIONS: { [name: string]: Region } = {};
+
+/**
+ * Information about a specific AWS region.
+ *
+ * Settings for a particular region can be obtained using `Region.named`.
+ * Settings can be registered or overridden using `Region.register`.
+ */
+export class Region {
+  /**
+   * Register a new region's settings. Use this to inject your own region settings.
+   *
+   * @param region    the new region settings to be registered.
+   * @param overwrite whether existing settings for the region should be overwritten.
+   */
+  public static register(region: Region, overwrite = true): void {
+    if (region.name in REGIONS) {
+      if (overwrite) {
+        process.stderr.write(`WARNING: overwriting existing region settings for ${region.name}\n`);
+      } else {
+        throw new Error(`Attempted to overwrite region settings for ${region.name}`);
+      }
+    }
+    REGIONS[region.name] = region;
+  }
+
+  /**
+   * Retrieve settings for a particular region name.
+   *
+   * @param region the name of the region being looked up.
+   */
+  public static named(region: string): Region {
+    return REGIONS[region] || new Region(region, {});
+  }
+
+  /** The partition of the AWS region (e.g: aws-cn) */
+  public readonly partition: string;
+
+  /** The domain name suffix for the partition */
+  public readonly domainSuffix: string;
+
+  /** Information about API Gateway's settings in the region */
+  public readonly apiGateway: ApiGatewaySettings;
+
+  /** Information about the behavior of S3 Website hosting in the region */
+  public readonly s3StaticWebsite: S3StaticWebsiteSettings;
+
+  /** Whether the AWS::CDK::Metadata CloudFormation resource is available */
+  public readonly hasCdkMetadataResource: boolean;
+
+  private readonly servicePrincipalMaker: IServicePrincipalMaker;
+
+  /**
+   * Initializes new Region settings, so they can be passed to `Region.register`.
+   *
+   * @param name The name of the AWS region (e.g: us-east-1)
+   */
+  public constructor(public readonly name: string, props: RegionProps) {
+    this.partition = props.partition || 'aws';
+    this.domainSuffix = props.domainSuffix || 'amazonaws.com';
+    this.hasCdkMetadataResource = props.hasCdkMetadataResource !== false;
+
+    const edgeHostedZoneId = props.apiGatewayEdgeOptimizedHostedZoneId == null
+      ? 'Z2FDTNDATAQYW2'
```

Comment:
This thing seems awfully specific to be a default value?

Reply:
I can only live by what the documentation says.